### PR TITLE
Add alpine build

### PR DESCRIPTION
**Pull request by seanmor5**
This is the correct Dockerfile+build for pre-compiling on Alpine. We have to patch TF and also patch the LLVM commit that TF uses, Lol

Will add the action next

### CODE REVIEW

**Comment by seanmor5** [comment]
I will try to upstream the alpine patch in TensorFlow so we do not need to keep this weird patch/branching logic

**Comment by jonatanklosko** [comment]
Looks good to me! If we can upstream, that would be perfect :D

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          apk add --no-cache libstdc++ openjdk11 && \
+          apk add --no-cache bash curl git wget && \
+          apk add --no-cache musl-dev make libexecinfo libexecinfo-dev && \
+          apk add --no-cache coreutils gcc g++ linux-headers unzip zip && \
+          apk add --no-cache automake gcc subversion && \
+          DIR=$(mktemp -d) && cd ${DIR} && \
+          curl -sLO https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-dist.zip && \
+          echo ${BAZEL_SHA256SUM} | sha256sum --check && \
+          unzip bazel-${BAZEL_VERSION}-dist.zip && \
+          EXTRA_BAZEL_ARGS="--tool_java_runtime_version=local_jdk" bash ./compile.sh && \
```
Comment (jonatanklosko) [inline .github/workflows/test.yaml:33]: I'd also try the setup-bazel action here, this way we don't need to compile it!

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          cp ${DIR}/output/bazel /usr/local/bin/ && \
+          rm -rf ${DIR}
+      - name: Install hex
+        run: mix local.hex --force
```
Comment (jonatanklosko) [inline .github/workflows/test.yaml:38]: Just in case :)

```suggestion
        run: mix local.hex --force && mix local.rebar --force
```

**Comment by seanmor5** [comment]
Yeah I think we would need to bump to a revision. Thinking about it though bumping to a revision would likely break this PR since this patches 2.11 and I am sure there are changes. So maybe instead we:

1. Release new XLA without ROCm
2. Try to upstream Alpine fixes to TF and see if we can get that included in TF2.12
3. If they are accepted we can bump to that revision and add a precompile step for ROCm

In the meantime, I think my PR for the conditional build will still work, and we just need to add a note in the README that to build for ROCm you need to manually set `TENSORFLOW_GIT_REV` and `XLA_BUILD=true` to build from source

**Comment by jonatanklosko** [comment]
Sounds good to me!

> In the meantime, I think my PR for the conditional build will still work

If those bazel targets are in 2.11 then this approach totally makes sense, and we figure out precompilation on the next release.

**Pull request merged**